spring-cloud-vault-config-databases/src/test/java/org/springframework/cloud/vault/config/databases/VaultConfigCouchbaseDatabaseTests.java

```diff
@@ -0,0 +1,139 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.vault.config.databases;
+
+import java.net.InetSocketAddress;
+
+import com.couchbase.client.java.*;
+import com.couchbase.client.java.kv.*;
+import com.couchbase.client.java.json.*;
+import com.couchbase.client.java.query.*;
+import com.couchbase.client.core.error.*;
+
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.time.Duration;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.vault.util.CanConnect;
+import org.springframework.cloud.vault.util.VaultRule;
+import org.springframework.cloud.vault.util.Version;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.vault.core.VaultOperations;
+
+import static org.junit.Assume.assumeTrue;
+
+/**
+ * Integration tests using the database secret backend. In case this test should fail
+ * because of SSL make sure you run the test within the
+ * spring-cloud-vault-config/spring-cloud-vault-config directory as the keystore is
+ * referenced with {@code ../work/keystore.jks}.
+ *
+ * @author Francis Hitchens
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = VaultConfigCouchbaseDatabaseTests.TestApplication.class,
+		properties = { "spring.cloud.vault.couchbase.enabled=true",
+				"spring.cloud.vault.couchbase.role=readonly",
+				"spring.data.couchbase.username=foo",
+				"spring.data.couchbase.password=bar",
+				"spring.main.allow-bean-definition-overriding=true" })
+public class VaultConfigCouchbaseDatabaseTests {
+
+	private static final int COUCHBASE_PORT = 8091;
+
+	private static final String COUCHBASE_HOST = "localhost";
+
+	@Value("${spring.data.couchbase.username}")
+	String username;
+
+	@Value("${spring.data.couchbase.password}")
+	String password;
+
+	Cluster cluster;
+
+	/**
+	 * Initialize the couchbase secret backend.
+	 */
+	@BeforeClass
+	public static void beforeClass() {
+
+		VaultRule vaultRule = new VaultRule();
+		vaultRule.before();
+
+		assumeTrue(CanConnect.to(new InetSocketAddress(COUCHBASE_HOST, COUCHBASE_PORT)));
+		assumeTrue(vaultRule.prepare().getVersion()
```

Comment:
See the previous file's version note for further details.

Reply:
From the Vault release notes I see this plugin infrastructure was added in 0.7.1 so I think this version should be good. You add the custom plugin outside of the vault version.